### DISCUSSION TITLE
add support for new device AduroSmart Scene controller

### DISCUSSION THREAD

**Discussion by bsamarketing** [Koenkk/zigbee2mqtt#11588] [Devices]
I am trying to add support for a the AduroSmart Scene Remote

https://www.vesternet.com/en-id/collections/zigbee-keyfobs-remotes-wall-controllers/products/zigbee-aduro-scene-remote

, but I am getting nowhere, and could use some help.

There is support for many of their other devices, so I am guessing it must be possible

https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/devices/adurosmart.js

I tried following this https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html#instructions

but I can’t get it to work. any help to get me started would would be most appreciated.

I am running zigbee2mqttin Homeassistant hass.io

**Comment by bsamarketing** [comment]
I have made some progress, and managed to add the device.

Can anyone tell me how to find out what the device exposes

I am guessing I need to add something to the exposes: line, and am guessing it’s e.action(), but what are the parameters? 

I tried exposes: e.action(‘on’,’off’). but a, just getting N/A reported.

Any suggestions what I am doing wrong?

**Comment by bsamarketing** [comment]
Managed to sort it. code for the device file is - with this as an external converter it’s all working

const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    zigbeeModel: ['ADUROLIGHT_CSC'], // The model ID from: Device with modelID 'lumi.sens' is not supported.
    model: 'Scene Remote', // Vendor model number, look on the device for a model number
    vendor: 'AduroSmart', // Vendor of the device (only used for documentation and startup logging)
    description: 'AduroSmart Remote Scene Controller', // Description of the device, copy from vendor site. (only used for documentation and startup logging)
    fromZigbee: [fz.battery,fz.command_toggle,fz.command_recall], // We will add this later
    toZigbee: [], // Should be empty, unless device can be controlled (e.g. lights, switches).
    exposes: [e.battery(), e.action('toggle','recall_253','recall_254','recall_255'), e.linkquality()], // Defines what this device exposes, used for e.g. Home Assistant discovery and in the frontend
};

module.exports = definition;

**Reply by Koenkk** [reply]
added

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)

**Comment by bsamarketing** [comment]
Thanks for your great work

Regards
Duncan
Duncan Wright MBA MCIM Chartered Marketer
***@***.******@***.***>

T: 01457 851111
W: www.bsamarketing.com<http://www.bsamarketing.com/>

BSA Marketing - Glossop Gasworks - Arundel Street - GLOSSOP SK13 7AB Co Reg No:3169628

Follow us on Twitter<https://twitter.com/bsamarketing> ***@***.***) or LinkedIn<https://www.linkedin.com/company/bsa-marketing> to receive regular marketing focused content
